### PR TITLE
build(deps): go 1.23, mockery 2.46.2, golangci-lint 1.61

### PR DESCRIPTION
**Pull request by lklimek**
## Issue being fixed or feature implemented

Go 1.23 is required by mockery 2.46.2 #944 

## What was done?

Upgraded Go to 1.23
Upgraded mockery to 2.46.2 and regenerated mocks
Upgraded golanci-lint to 1.61

## How Has This Been Tested?

GHA

## Breaking Changes

Go 1.23 is required to build

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

## Release Notes

- **New Features**
  - Updated the minimum Go version requirement to 1.23 across various components, enhancing compatibility and performance.

- **Documentation**
  - Revised tutorials and README files to reflect the updated Go version and provide clearer instructions for creating Tendermint Core applications.

- **Bug Fixes**
  - Improved workflow configurations in GitHub Actions to ensure consistent use of the updated Go version across all jobs.

- **Chores**
  - Updated dependencies in the project to their latest compatible versions, ensuring better stability and functionality.

### CODE REVIEW

**Comment by lklimek** [comment]
Failing tests:

* check-mocks  - will be fixed in #955 
* golangci-lint - will be also fixed in #956
*

File: go.mod [hunk 1]
```diff
@@ -163,93 +173,85 @@
...
-	github.com/golangci/lint-1 v0.0.0-20191013205115-297bf364a8e0 // indirect
-	github.com/golangci/maligned v0.0.0-20180506175553-b1d89398deca // indirect
-	github.com/golangci/misspell v0.4.1 // indirect
-	github.com/golangci/revgrep v0.5.2 // indirect
-	github.com/golangci/unconvert v0.0.0-20180507085042-28b1c447d1f4 // indirect
-	github.com/gordonklaus/ineffassign v0.0.0-20230610083614-0e73809eb601 // indirect
+	github.com/golangci/gofmt v0.0.0-20240816233607-d8596aa466a9 // indirect
+	github.com/golangci/misspell v0.6.0 // indirect
+	github.com/golangci/revgrep v0.5.3 // indirect
+	github.com/golangci/unconvert v0.0.0-20240309020433-c5143eacb3ed // indirect
```
[the remark this answers is not in the document]
Comment (lklimek) [reply go.mod:215]: These are indirect dependencies, we don't control that.

File: internal/p2p/mocks/transport.go [hunk 2]
```diff
@@ -48,7 +45,7 @@ func (_m *Transport) Accept(_a0 context.Context) (p2p.Connection, error) {
 }
 
 // AddChannelDescriptors provides a mock function with given fields: _a0
-func (_m *Transport) AddChannelDescriptors(_a0 []*conn.ChannelDescriptor) {
+func (_m *Transport) AddChannelDescriptors(_a0 []*p2p.ChannelDescriptor) {
```
[the remark this answers is not in the document]
Comment (lklimek) [reply internal/p2p/mocks/transport.go:48]: conn.ChannelDescriptor and p2p.ChannelDescriptor  are aliases

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**